assets/prometheus-k8s/rules.yaml

```diff
@@ -335,60 +332,6 @@ spec:
       record: cluster:node_instance_type_count:sum
     - expr: sum(etcd_object_counts) BY (instance)
       record: instance:etcd_object_counts:sum
-    - expr: sum(((cluster:master_infra_nodes * on(node) group_left(label_node_hyperthread_enabled)
-        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled!="true"})
-        or (cluster:master_infra_nodes * on(node) group_left(label_node_hyperthread_enabled)
-        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
-        * on(node) group_left kube_node_status_capacity_cpu_cores) or on(node) ((cluster:master_nodes
-        * on(node) group_left(label_node_hyperthread_enabled) cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled!="true"})
-        or (cluster:master_nodes * on(node) group_left(label_node_hyperthread_enabled)
-        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
-        * on(node) group_left kube_node_status_capacity_cpu_cores) or on(node) ((cluster:infra_nodes
-        * on(node) group_left(label_node_hyperthread_enabled) cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled!="true"})
-        or (cluster:infra_nodes * on(node) group_left(label_node_hyperthread_enabled)
-        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
-        * on(node) group_left kube_node_status_capacity_cpu_cores) or on(node) (cluster:hyperthread_enabled_nodes
-        * on(node) group_left kube_node_status_capacity_cpu_cores) or on(node) ((kube_node_labels
-        unless on(node) cluster:hyperthread_enabled_nodes) * on(node) group_left kube_node_status_capacity_cpu_cores))
-        BY (label_node_openshift_io_os_id, label_kubernetes_io_arch, label_node_hyperthread_enabled,
-        label_node_role_kubernetes_io_master_infra, label_node_role_kubernetes_io_master,
-        label_node_role_kubernetes_io_infra)
-      record: node_role_os_version_machine:cpu_capacity_cores:sum
```

Comment:
We have to preserve the old name for consumers - we are already using this in layers on top of OCM.

Reply:
The new metrics provide basically the same info minus the hyperthreading label, since it can be determined from the number of cores vs. cpus.  So I could rename the new metrics to match these names if that would prevent issues in the higher layers.

Reply:
Hrm, are we sure that actually captures the use case?  We need the hyper threading label to be super obvious so we can account appropriately.  Is the argument here that we *can* query it separately, or that we *must* query it separately?

Reply:
I think we don't need to query hyperthreading at all.  I discussed it a bit with @luciddreamz and IIUC we just needed to know the number of sockets, cores, and cpus.  The idea to get hyperthreading info was just an indirect way to get the number of cores.

Reply:
There also might be an edge case where you could have two sockets on the same node, one with hyperthreading and one without.  In that case the hyperthreading label would be inaccurate.